src/compile/scale/domain.ts

```diff
@@ -408,10 +409,45 @@ export function mergeDomains(domains: VgNonUnionDomain[]): VgDomain {
  * Return `undefined` otherwise.
  *
  */
-export function getFieldFromDomains(domains: VgNonUnionDomain[]): string {
-  const domain = mergeDomains(domains);
+export function getFieldFromDomain(domain: VgDomain): string {
   if (isDataRefDomain(domain) && isString(domain.field)) {
     return domain.field;
+  } else if (isDataRefUnionedDomain(domain)) {
+    let field;
+    for (const nonUnionDomain of domain.fields) {
+      if (isDataRefDomain(nonUnionDomain) && isString(nonUnionDomain.field)) {
+        if (!field) {
+          field = nonUnionDomain.field;
+        } else if (field !== nonUnionDomain.field) {
+          log.warn('Detected faceted independent scales that union domain of multiple fields from different data sources.  We will use the first field.  The result cell size may be incorrect.');
```

Comment:
@domoritz I kinda feel obligated to throw warning when something might be incorrect, but this warning also looks scary.  Any thoughts?

Reply:
I think this should be fine given the dataflow will be optimized for this case soon.